src/Connection.js

```diff
@@ -7,6 +7,24 @@ module.exports = class Connection extends events.EventEmitter {
         this.id = socket.id
         this.socket = socket
         this.streams = []
+        this.preferredPayloadVersion = Connection.getVersionFromUrl(socket.upgradeReq.url)
+    }
+
+    static getVersionFromUrl(url) {
+        const parts = url.split('?')
```

Comment:
There definitely exists some minimal library for parsing query params, let's just bring it in instead of reimplementing it.